client/landing/gutenboarding/index.tsx

```diff
@@ -129,11 +123,47 @@ window.AppBoot = async () => {
 					</Route>
 				</Switch>
 			</BrowserRouter>
-		</I18nProvider>,
+		</CalypsoI18nProvider>,
 		document.getElementById( 'wpcom' )
 	);
 };
 
+const CalypsoI18nProvider: React.FunctionComponent< {
+	initialLocaleData: any;
+} > = ( { children, initialLocaleData } ) => {
+	const [ localeData, setLocale ] = React.useState( initialLocaleData );
+
+	// For testing
+	React.useEffect( () => {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+		// @ts-ignore
+		window.updateLocale = async ( newLocale: string ) => {
```

Comment:
To make this suitable for building on in #40011, just keeping window.updateLocale would not be ideal.

I'm not sure of the best mechanism 🤔 , do you have ideas about the final form of this?

Reply:
👍  `window.updateLocale` was _only_ intended for testing. This started as a way to verify reactive i18n changes were working as expected. This PR can be abandoned entirely if that makes sense, it does happen to contain patterns that may serve as inspiration for plans outlined in #42836 and the language switcher in #40011.

For example, a provider might expose a callback that would trigger language switching which would replace `window.updateLocale`.

Reply:
hmm yea ok, yea so it sounds like we can use this PR for inspiration and re-implement some parts of it when working on #40011. Then eventually abandon this PR once its patterns have been incorporated into #40011